src/wp-includes/blocks.php

```diff
@@ -760,6 +760,108 @@ function get_hooked_blocks( $name ) {
 	return $hooked_blocks;
 }
 
+function insert_hooked_blocks( $block ) {
```

Comment:
Can we land `insert_hooked_block` and tackle `insert_hooked_blocks` in a follow-up? Well, I should start by saying that I don't fully understand the difference between both 😄

Reply:
> Can we land `insert_hooked_block` and tackle `insert_hooked_blocks` in a follow-up?

Yes! I was actually thinking the exact same thing! 😄 

> Well, I should start by saying that I don't fully understand the difference between both 😄 

Hehe, I can't blame you, there's a lot of moving parts. I promise it'll all make sense soon! 😊

Reply:
940c24bb14

Reply:
To explain a bit right away:

`insert_hooked_block` is fairly low-level, and accepts a hooked block for insertion, plus the insertion point specified by the anchor block and relative position (and possibly parent block and block/chunk indices, [see](https://github.com/WordPress/wordpress-develop/pull/5240#issuecomment-1724961426)).

`insert_hooked_blocks` is more high-level: Given a block (found in a template or pattern), it will look up all blocks hooked into it (via `get_hooked_blocks`) and then call `insert_hooked_block` to insert them in the correct locations.

HTH!